src/BinaryBuilder.jl

```diff
@@ -209,4 +210,11 @@ function versioninfo()
     end
 end
 
+function url_hash(url::AbstractString)
+    path = download(url)
```

Comment:
Should clean up the downloaded file? Or is the side effect intended?

Another way of writing this is to just have it as `hash(file)` and have users call it with e.g. `hash(download(...))`.

Reply:
oh yeah, it should clean up the downloaded file.

I think the common use-case is you have a URL you want to add and you need the hash, so I prefer the API as-is.

It might be nice to tie into the other download machinery in BB that shows the nice progress bar though. I just used this on a large-ish tarball and it just sits there for a bit while it downloads, which could be surprising for a user who's not really thinking about what's going on under the hood.

Reply:
You can have it be auto-cleaned up by downloading within a temporary prefix (which will get auto-deleted after you exit the `do` block).  You can also use `BinaryProvider.download(url, path; verbose=true)` to get the neat download bar.  Putting that together, what I'm suggesting looks like this:

```
function url_hash(url::AbstractString)
    temp_prefix() do p
        local_path = joinpath(p, basename(url))
        BinaryProvider.download(url, local_path; verbose=true)
        return hex2bytes(sha256(open(local_path)))
    end
end
```